packages/swapper/src/types.ts

```diff
@@ -456,6 +489,7 @@ export type CheckTradeStatusInput = {
   accountId: AccountId | undefined
   stepIndex: SupportedTradeQuoteStepIndex
   config: SwapperConfig
+  swap: Swap | undefined
```

Comment:
q: shouldn't this be guaranteed to be define? i.e we should have a swap when we hit this, either idle, pending or completed/failed, and shouldn't call this without one?

Reply:
checkTradeStatus is also used in the rfox bridge without a quoteId, for this particular case we don't have a swap (yet?) 

That's the reason why it's not defined all the time, but I agree this isn't practical and would maybe require more at some points